lib/main.dart

```diff
@@ -76,13 +75,11 @@ class _StartScreenState extends State<StartScreen> {
         return AppUtility.widgetForAsyncSnapshot(
           snapshot: snapshot,
           onLoading: () => const LoadingScreen(),
-          onError: () => SignInScreen(
-            nextRouteName: widget.nextRouteName,
-          ),
+          onError: () => const WelcomeScreen(),
           onReady: () {
             if (snapshot.data?.response == null) {
               // Empty result means that the user not signed in.
-              return SignInScreen(nextRouteName: widget.nextRouteName);
+              return const WelcomeScreen();
```

Comment:
Here the nextRoutName param is lost. Lucas added this param so that users could open a screen directly from a notification. If the user is logged in, this _if_ statement is skipped and the WidgetsBinding callback below will make sure to redirect them to the corresponding page. However if the user is not logged in, this condition is executed to have them sign in and THEN redirect them to the nextRoute (which could be anything). 

By removing the nextRouteName param here the user will go through the sign in flow as usual and always end up in the home page. This is not super serious but I wanted to point this out since it renders the SignInScreen nextRoutName meaningless.

I would suggest something like this, but feel free to solve it however you wish:

```
if (snapshot.data?.response == null) {
    // Empty result means that the user not signed in.
    return widget.nextRouteName == Routes.home.name? 
      const WelcomeScreen() 
      : SignInScreen(nextRouteName: widget.nextRouteName;
}
```

This means: if the next route is home (opening the app normally) show the WelcomeScreen, otherwise the user just needs to sign-in to be redirected to a different page (coming from clicking a notification, or whatever).

Reply:
Thank you for the explanation! I updated it to use the signin screen when nextRoutName is available.